### PR TITLE
allow conan install pkg/[*]@user/channel

### LINKED ISSUE #5905: wide-open version range on command-line now misinterpreted as path
In the provisioning scripts to create jenkins slaves, I have several commands of the style

`conan install name/[*]@user/channel --profile foo` which are intended to pre-download the latest available version of a package to be baked in to the VM disk image, so that it will not have to be re-extracted by each job (the compute resources are launched ephemerally).

For a few of these packages, there is also some action in the deploy() method (e.g. registering with the CMake package registry to bootstrap find_package(cmake-conan).

I just updated these provisioning scripts to conan 1.19, and such commands seem to no longer work; they are mis-interpreted as file paths, and now fail with 

> ERROR: Conanfile not found at ERROR: Conanfile not found at $PWD/name/[*]@user/channel

Instead of the expected 

> ERROR: Version range '*' from requirement 'package/[*]@user/channel' required by 'virtual' could not be resolved in local cache

(or of course success, if I had used a real name/user/channel instead of bogus placeholders).  But the error that this is being treated as a file paths rather than a references is visible even with no real package. However, this is not happening with version ranges in general, just '*': otherwise-equivalent forms like `conan install package/[x]@user/channel --profile foo` or `conan install package/[>=0]@user/channel --profile foo` seem to work fine.

searching among the releases on pypi, this seems to have broken between conan 1.17.2 and 1.18.0, so I suspect some change to parsing with the changes that made user/channel optional.

It's not a big deal to just change these scripts to use `[x]` instead of `[*]`, but it seems likely to have been unintentional (if it was on purpose, maybe https://docs.conan.io/en/latest/versioning/version_ranges.html should warn against the use of '*' in semver ranges?).

I see the same behavior on both windows 10 (python 3.7.4) and linux (ubuntu bionic, python 3.6.8).

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Bugfix: Allow ``conan install pkg/[*]@user/channel`` resolving to a reference, not a path.
Docs: Omit

Fix #5905

### CODE REVIEW

**Comment by puetzk** [comment]
I think this PR would still leave things like `pkg/[1.*]@user/channel` broken? I don't have any usages like that, but they're equally valid syntax for an x-range: https://github.com/npm/node-semver#x-ranges-12x-1x-12-

**Comment by puetzk** [comment]
Thanks!

**Comment by memsharded** [comment]
> I think this PR would still leave things like pkg/[1.*]@user/channel broken? I don't have any usages like that, but they're equally valid syntax for an x-range: 

Fixed too.

What we will definitely do is in Conan 2.0 to make the arguments non-ambiguous: a path will be explicitly a path, and a package reference will be explicitly a package reference

**Comment by puetzk** [comment]
Yeah, the fact that conan install has such a split personality can be quite confusing (`conan install <path>` means "download the requirements listed in this conanfile.txt or conanfile.py and run generators) and `conan install <ref>` means "download or build this package and run deploy()".

I have no problem with some breaking changes to disambiguate this command, especially in a major version. I can adjust my scripts accordingly when upgrading. This change just seemed inadvertent.

File: conans/test/unittests/model/ref_test.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ class CheckValidRefTest(unittest.TestCase):
     def test_string(self):
         self.assertTrue(check_valid_ref("package/1.0@user/channel"))
         self.assertTrue(check_valid_ref("package/1.0@user/channel"))
+        self.assertFalse(check_valid_ref("package/*@user/channel"))
```
Comment (lasote) [inline conans/test/unittests/model/ref_test.py]: Check here the version ranges too (and with *)

Comment (memsharded) [reply conans/test/unittests/model/ref_test.py]: Done. Indeed the ``check_valid_ref`` is weird.

Comment (lasote) [reply conans/test/unittests/model/ref_test.py]: And is that a reason to not test it or what? :P

Comment (memsharded) [reply conans/test/unittests/model/ref_test.py]: Nop, definitely tests were needed.
I am saying that I really want to get rid of all this mess, looking forward to Conan 2.0.

Comment (lasote) [reply conans/test/unittests/model/ref_test.py]: agree

**Pull request merged**